ai/WFLY-19381-[EXPERIMENTAL]-Provide_a_galleon_feature_pack_to_facilitate_Genrative_AI_development.adoc

```diff
@@ -0,0 +1,264 @@
+ We should provide layers to provision the server according to the needs.
+
+
+=== Nice-to-Have Requirements
+
+ * Provide annotations to be able to create AIServices using our configured resources.
+ * Replace the HTTP clients used so that we have only one that is supported for every API.
+ * Replace the JSON marshalling/umarshalling librairies so that we have only one (through RESTEasy) that is supported.
+ * Support for @Tool
+ * Adding support for ChatMemory
+ * Adding support for more APIs
+
+=== Non-Requirements
+// Use this section to explicitly discuss things that readers might think are required
+// but which are not required.
+
+=== Future Work
+// Use this section to discuss requirements that are not addressed by this proposal
+// but which may be addressed in later proposals.
+
+== Backwards Compatibility
+
+// Does this enhancement affect backwards compatibility with previously released
+// versions of WildFly?
+// Can the identified incompatibility be avoided?
+
+=== Default Configuration
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+The required librairies should be added automatically on the deployment classpath. 
+
+=== Interoperability
+
+== Implementation Plan
+
+=== Embeddings models
+
+The extension should provide resources to define `dev.langchain4j.model.embedding.EmbeddingModel`.
+
+It should expose a simple `embedding-model` resource with the following attributes:
+
+* module: the jboss module containing the code of the model.
+* embedding-class: the name of the class to use to load the model.
+
+----
+/subsystem=ai/embedding-model=myembedding:add(module=dev.langchain4j.embeddings.all-minilm-l6-v2, embedding-class=dev.langchain4j.model.embedding.AllMiniLmL6V2EmbeddingModel)
+----
+
+It should also support LLM backed embedding models like for ollama for example.
+We should have an `ollama-embedding-model` resource with the following attributes:
+
+* base-url: endpoint to connect to an Ollama chat model.
+* connect-timeout: timeout for the Ollama embedding model.
+* log-requests: enabling the tracing of requests going to Ollama.
+* log-responses: enabling the tracing of responses from Ollama.
+* model-name: the name of the embedding model served by Ollama.
+
+----
+subsystem=ai/ollama-embedding-model=myembedding:add(base-url="http://192.168.1.11:11434", model-name="llama3:8b")
+----
+
+=== Embeddings stores
+
+The extension should provide resources to define `dev.langchain4j.store.embedding.EmbeddingStore`.
+
+It should expose a simple `in-memory-embedding-store` resource with the following attributes:
+
+* file: the file to load the in memory embedding store content from.
+
+----
+/subsystem=ai/in-memory-embedding-store=mystore:add(file=/home/ai/dev/wildfly-admin-embeddings.json)
+----
+
+It should also support vector database backed embedding store like for Weaviate.
+It should expose a simple `weaviate-embedding-store` resource with the following attributes:
```

Comment:
Followup on my concern expressed above. For example there is a lot of relevant vector dbs out there (chroma, qdrant, milvus, pinecone, elasticsearch, ...) https://lakefs.io/blog/12-vector-databases-2023/ I am questioning if providing resources for all of them does scale?

Reply:
We don't have to support everyone of them, all the more so as each bring their own dependency tree